From e3e8afa5776767af5dc4ca874d159a8660a90e5d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alex=20R=C3=B8nne=20Petersen?= <alex@alexrp.com>
Date: Thu, 18 Sep 2025 00:11:21 +0200
Subject: [PATCH 078/133] test: disable some SIMD ABI tests on LoongArch
X-Developer-Signature: v=1; a=openpgp-sha256; l=1182; i=xtex@astrafall.org;
 h=from:subject; bh=XLSVl4R9h8YaQa0mpS/OOaKBXB+m7BhgWN2iIzXpFmc=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78thq9I4d/BvPr3jgbzR7125RQwPViC9B1SELPa/tv
 cbfy3i3YwILgxgXg6WYIkuRYYM3q046v+iyclmYOaxMIEOkRRoYGBgYWBj4chPzSo10jPRMtQ31
 DA11jHSMGLg4BWCq5fQYfuxq1oi3nmdm97czWGrVw5gLMQWZ08PnzTbtenQ+vn1CLcN/vyVBOyc
 weMnaX2Znuy+iu5mRtUSscHv+CoXrNStE185kAgA=
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

(cherry picked from commit f1c2ae271cd0e9caac332a456ff8dbd2fc4c277a)
---
 test/c_abi/main.zig | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/test/c_abi/main.zig b/test/c_abi/main.zig
index d32307e485c7..73e9c5c8dc90 100644
--- a/test/c_abi/main.zig
+++ b/test/c_abi/main.zig
@@ -1125,6 +1125,7 @@ extern fn c_ret_medium_vec() MediumVec;
 test "medium simd vector" {
     if (builtin.zig_backend == .stage2_x86_64 and !comptime builtin.cpu.has(.x86, .avx)) return error.SkipZigTest;
     if (builtin.cpu.arch.isPowerPC64()) return error.SkipZigTest;
+    if (builtin.cpu.arch.isLoongArch()) return error.SkipZigTest;
 
     c_medium_vec(.{ 1, 2, 3, 4 });
 
@@ -1145,6 +1146,7 @@ test "big simd vector" {
 
     if (builtin.cpu.arch.isMIPS64() and builtin.mode != .Debug) return error.SkipZigTest;
     if (builtin.cpu.arch.isPowerPC64()) return error.SkipZigTest;
+    if (builtin.cpu.arch.isLoongArch()) return error.SkipZigTest;
     if (builtin.zig_backend == .stage2_llvm and builtin.cpu.arch == .x86_64 and builtin.os.tag == .macos and builtin.mode != .Debug) return error.SkipZigTest;
 
     c_big_vec(.{ 1, 2, 3, 4, 5, 6, 7, 8 });
-- 
2.52.0

